internal/util/util.go

```diff
@@ -236,13 +236,21 @@ func IsVolumeSnapshotHasVSCDeleteSecret(vs *snapshotv1beta1api.VolumeSnapshot) b
 	return nameExists && nsExists
 }
 
-// AddAnnotations adds the supplied key-values to the annotations on the object
-func AddAnnotations(o *metav1.ObjectMeta, vals map[string]string) {
-	if o.Annotations == nil {
-		o.Annotations = make(map[string]string)
+func addToMap(cur *map[string]string, vals map[string]string) {
+	if *cur == nil {
+		*cur = make(map[string]string)
 	}
-
 	for k, v := range vals {
-		o.Annotations[k] = v
+		(*cur)[k] = v
 	}
 }
+
+// AddAnnotations adds the supplied key-values to the annotations on the object
+func AddAnnotations(o *metav1.ObjectMeta, vals map[string]string) {
+	addToMap(&o.Annotations, vals)
```

Comment:
I would probably drop the `addToMap` helper and just duplicate the relevant code in both `AddAnnotations` and `AddLabels` - I get why you're doing the pointer stuff with the maps but I think it's probably simpler to avoid it and just duplicate the few lines of code.